### PR TITLE
Groups enabled and group categories endpoint on JS Config

### PR DESCRIPTION
**Pull request by marcospri**
Adds 
``` 
"filePicker": {
  "groupsEnabled":false,       
  "listGroupCategories":{
    "authUrl":"http://localhost:8001/api/canvas/oauth/authorize",
    "path":"/api/canvas/courses/125/group_categories"
  },
}
```

to JS config when in `content-item-selection` mode

Full config opening the content selection modal on local:

```{
   "api":{
      "authToken":"Bearer ey"
   },
   "canvas":{
      
   },
   "debug":{
      "tags":[
         "role:instructor"
      ]
   },
   "dev":true,
   "filePicker":{
      "blackboard":{
         "enabled":false,
         "listFiles":{
            "authUrl":"http://localhost:8001/api/blackboard/oauth/authorize",
            "path":"/api/blackboard/courses/XXXX/files"
         }
      },
      "canvas":{
         "enabled":true,
         "groupsEnabled":false,
         "listFiles":{
            "authUrl":"http://localhost:8001/api/canvas/oauth/authorize",
            "path":"/api/canvas/courses/125/files"
         },
         "listGroupCategories":{
            "authUrl":"http://localhost:8001/api/canvas/oauth/authorize",
            "path":"/api/canvas/courses/125/group_categories"
         },
         "ltiLaunchUrl":"http://localhost:8001/lti_launches"
      },
      "formAction":"https://hypothesis.instructure.com/courses/125/external_content/success/external_tool_dialog",
      "formFields":{
         "lti_message_type":"ContentItemSelection",
         "lti_version":"LTI-1p0"
      },
      "google":{
         "clientId":"XXXX",
         "developerKey":"XXXX",
         "origin":"hypothesis.instructure.com"
      },
      "vitalSource":{
         "enabled":false
      }
   },
   "mode":"content-item-selection"
}
```

### CODE REVIEW

File: lms/resources/_js_config.py [hunk 1]
```diff
@@ -144,6 +144,8 @@ def enable_lti_launch_mode(self):
             "allowedOrigins": self._request.registry.settings["rpc_allowed_origins"]
         }
 
+        self._canvas_settings()
```
Comment (robertknight) [inline lms/resources/_js_config.py:147]: This section of the config is used when launching an assignment, whereas the information about whether groups are enabled and the metadata to make an API call to list group categories are needed in assignment selection mode, known as "content item selection mode" in the context of this module. I believe that terminology comes from the LTI spec. See the `enable_content_item_selection_mode` method.

File: lms/static/scripts/frontend_apps/config.js [hunk 2]
```diff
@@ -61,8 +61,10 @@ import { createContext } from 'preact';
  * @prop {Object.<string,string>} formFields
  * @prop {Object} canvas
  *   @prop {boolean} canvas.enabled
+ *   @prop {boolean} canvas.groupsEnabled
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/config.js:64]: Added this here, couldn't find other references to `FilePickerConfig` but not sure  if needed somewhere else

**Review by robertknight — APPROVED** [review]
LGTM. I've built https://github.com/hypothesis/lms/pull/2687 on top of this. We did agree to change the terminology from "group categories" to "group sets" everywhere per https://hypothes-is.slack.com/archives/C1MA4E9B9/p1621000658481700?thread_ts=1620638554.378200&cid=C1MA4E9B9. I'm not sure if you wanted to make that change in this PR or a follow-up? I don't mind either way.

File: tests/unit/lms/resources/_js_config_test.py [hunk 3]
```diff
@@ -87,17 +92,6 @@ def test_it_doesnt_enable_the_canvas_file_picker_if_the_lms_isnt_Canvas(
 
         self.assert_canvas_file_picker_not_enabled(js_config)
 
-    def test_it_doesnt_enable_the_canvas_file_picker_if_the_consumer_key_isnt_found_in_the_db(
```
Comment (robertknight) [inline tests/unit/lms/resources/_js_config_test.py:90]: Per @marcospri in Slack:

> Yep, [this change is] also done in another PR from Sean https://github.com/hypothesis/lms/pull/2691 and I didn't want to do a different fix and then rebase the deletion

**Review by seanh — APPROVED** [review]

**Pull request merged**